packages/client/src/sql/parameters.ts

```diff
@@ -0,0 +1,78 @@
+import { SQLQuery } from '.';
+import { isDefined, isObject, isString, isStringArray } from '../util/lang';
+
+function escapeElement(elementRepresentation: string) {
+  const escaped = elementRepresentation.replace(/\\/g, '\\\\').replace(/"/g, '\\"');
+
+  return '"' + escaped + '"';
+}
+
+function arrayString<T extends any[]>(val: T) {
+  let result = '{';
+  for (let i = 0; i < val.length; i++) {
+    if (i > 0) {
+      result = result + ',';
+    }
+    if (val[i] === null || typeof val[i] === 'undefined') {
+      result = result + 'NULL';
+    } else if (Array.isArray(val[i])) {
+      result = result + arrayString(val[i]);
+    } else if (val[i] instanceof Buffer) {
+      result += '\\\\x' + val[i].toString('hex');
+    } else {
+      result += escapeElement(prepareValue(val[i]));
+    }
+  }
+  result = result + '}';
+  return result;
+}
+
+// Code distilled from pg/lib/utils.js
```

Comment:
Would it be possible to link to this code? Does this code come with tests (or could we write our own), it feels very testable / edge-casey.

Reply:
Other libraries are using the same helper, also it's not a direct copy paste of a function, instead it tries to mimick the logic within that file.